packages/aws-cdk-lib/aws-stepfunctions/lib/private/json-path.ts

```diff
@@ -142,7 +149,9 @@ export function recurseObject(obj: object | undefined, handlers: FieldHandlers,
 
   const ret: any = {};
   for (const [key, value] of Object.entries(obj)) {
-    if (typeof value === 'string') {
+    if (value === NULL_SUBSTITUTE) {
+      Object.assign(ret, handlers.handleNull(key, null));
```

Comment:
Does the following not work?
```
if (value === null || value === undefined) {
     Object.assign(ret, handlers.handleNull(key, null));
}
```

Reply:
I think users want to specifically be able to distinguish `null` from `undefined` in typescript, but at the same time the two are the same in other languages like Python (`None`) so a magic value of some sort will satisfy both cases.

Reply:
@paulhcsun's right. This isn't an issue in TS (at least not as far as I know), but in jsii (languages) which cannot distinguish between `null` and `undefined`. But ASL (as a JSON-ish schema) sometimes requires users to use `null`